src/services/symbolDisplay.ts

```diff
@@ -41,7 +41,7 @@ namespace ts.SymbolDisplay {
         if (typeChecker.isArgumentsSymbol(symbol)) {
             return ScriptElementKind.localVariableElement;
         }
-        if (location.kind === SyntaxKind.ThisKeyword && isExpression(location)) {
+        if (location.kind === SyntaxKind.ThisKeyword && isExpression(location) || isThisInTypeQuery(location)) {

```

Comment:
This is probably out of scope for this review, but it would be nice for these two checks to use the *same* code (not `isExpression` vs `isExpressionContext`), and then be extracted to a function.